azurelinuxagent/common/conf.py

```diff
@@ -503,15 +503,18 @@ def get_monitor_network_configuration_changes(conf=__conf__):
     return conf.get_switch("Monitor.NetworkConfigurationChanges", False)
 
 
-def get_download_new_agents(conf=__conf__):
+def get_auto_update_to_latest_version(conf=__conf__):
     """
-    If True, the agent go through update logic to look for new agents to download otherwise it will stop agent updates.
-    NOTE: AutoUpdate.Enabled controls whether the Agent downloads new update and also whether any downloaded updates are started or not, while DownloadNewAgents controls only the former.
-    AutoUpdate.Enabled == false -> Agent preinstalled on the image will process extensions and will not update (regardless of DownloadNewAgents flag)
-    AutoUpdate.Enabled == true and DownloadNewAgents == true, any update already downloaded will be started, and agent look for future updates
-    AutoUpdate.Enabled == true and DownloadNewAgents == false, any update already downloaded will be started, but the agent will not look for future updates
+    If set to True, agent will update to the latest version
+    NOTE:
+        when turned on, both AutoEnabled.Enabled and UpdateToLatestVersion same meaning: update to latest version
+        when turned off, AutoEnabled.Enabled: reverts to pre-installed agent, UpdateToLatestVersion: uses latest version installed on the vm
+        Since we are deprecating AutoEnabled.Enabled, we need to honor if existing flag explicitly set to 'n' for backward compatibility.
     """
-    return conf.get_switch("Debug.DownloadNewAgents", True)
+    auto_update_enabled = get_autoupdate_enabled()
+    if not auto_update_enabled:
```

Comment:
Does the wiki mention that if AutoUpdate.Enabled is 'n', then AutoUpdate.UpdateToLatestVersion is ignored? If not, I think it should

Reply:
I added AutoUpdate.UpdateToLatestVersion applicable only if autoupdate set to 'y'. 

[maddieford](https://github.com/maddieford) [yesterday](https://github.com/Azure/WALinuxAgent/pull/3020#discussion_r1464202733)
Should we mention here it only applies when AutoUpdate.Enabled is 'y' OR not in the conf at all?

Member
@[maddieford](https://github.com/maddieford) maddieford [yesterday](https://github.com/Azure/WALinuxAgent/pull/3020#discussion_r1464202962)
Or alternatively, it is only applied if AutoUpdate.Enabled is not 'n'

Reply:
From this line it looks like if AutoUpdate.UpdateToLatestVersion is 'y' and AutoUpdate.Enabled is 'no' we return False here, is that correct?
In that case, I think we should add to the wiki that we ignore AutoUpdate.UpdateToLatestVersion if AutoUpdate.Enabled is 'n', even if AutoUpdate.UpdateToLatestVersion is 'y'